flytekit/clients/raw.py

```diff
@@ -96,6 +97,46 @@ def handler(*args, **kwargs):
     return handler
 
 
+def _handle_rpc_error_with_retries(fn):
+    def handler(*args, **kwargs):
+        """
+        Wraps rpc errors as Flyte exceptions and handles authentication the client.
+        Retries most errors. Avoid using for non-idempotent invocations.
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        retries = 3
+        max_wait_time = 1000
+
+        try:
+            for i in range(retries):
+                try:
+                    return fn(*args, **kwargs)
+                except _RpcError as e:
+                    if e.code() == _GrpcStatusCode.UNAUTHENTICATED:
```

Comment:
what's this special case for UNAUTHENTICATED?

Reply:
This already exited in the legacy handler above, keeping it as is. I believe the diff is no backoff here, not sure why. Would let @wild-endeavor  comment on this

Reply:
this is what triggers the code to attempt to authenticate, either by popping up a browser in the flyte-cli case, or by using a username password in the pyflyte case.